### PR TITLE
Add Group UUID Filtering for Components

### PR DESCRIPTION
**Pull request by CaprinaeWT**
## Pull request type

Jira Link: [INT-](url)

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Other (please describe):

## How to test this PR

1. Run the command to test the new group UUID filtering feature:
`./cli.js sync --type components --source <source-space-id> --target <target-space-id> --c-groups <group-uuid-1>,<group-uuid-2>,...` 

Please note that the --c-groups flag enables the filter based on the provided group UUIDs. To test the feature, replace source-value and target-value with the actual values of the source and target.

2. Verify that only the components belonging to the specified group UUIDs are synchronized between the source and target. Any components that do not belong to the specified group UUIDs should not be synchronized.

3. Try different combinations of group UUIDs to verify that the filter works as expected.

Thank you for testing the new group UUID filtering feature. Your feedback is highly appreciated.

## What is the new behaviour?

This pull request introduces a new feature that allows for filtering components based on their group UUID. With this new feature, users can specify one or more group UUIDs and synchronise only the components that belong to those groups.

The new feature is especially useful in scenarios where users need to manage a large number of components across different groups and need a way to easily synchronise components based on group membership.

The implementation includes a new filtering logic, which supports filtering based on the provided group UUIDs. The feature is also fully documented, with clear instructions on how to use the new filtering options.

Thank you for your attention and consideration.

### CODE REVIEW

File: src/cli.js [hunk 1]
```diff
@@ -289,9 +290,12 @@ program
         filter,
         keys,
         operations,
-        values
+        values,
+        cGroups
       } = options
 
+      const _cGroups = cGroups ? cGroups.split(',') : null
```
Comment (ademarCardoso) [inline src/cli.js]: I think it's better to change to `componentsGroups`, it is better for readability in the code.

File: src/tasks/sync-commands/components.js [hunk 2]
```diff
@@ -18,6 +18,7 @@ class SyncComponents {
     this.oauthToken = options.oauthToken
     this.client = api.getClient()
     this.presetsLib = new PresetsLib({ oauthToken: options.oauthToken, targetSpaceId: this.targetSpaceId })
+    this.cGroups = options.cGroups
```
Comment (ademarCardoso) [inline src/tasks/sync-commands/components.js]: Same thing here, `this.componentsGroups`

File: src/tasks/sync.js [hunk 3]
```diff
@@ -18,6 +18,7 @@ const SyncSpaces = {
     this.startsWith = options.startsWith
     this.filterQuery = options.filterQuery
     this.client = api.getClient()
+    this.cGroups = options._cGroups
```
Comment (ademarCardoso) [inline src/tasks/sync.js]: Same here

**Comment by ademarCardoso** [comment]
Hi @CaprinaeWT you added `;` in a lot of places in your last commit `c93789a`, please, revert this changes 😄 , we don't use `;` at the end of the lines. Thanks

**Comment by CaprinaeWT** [comment]
Hey @ademarCardoso, My bad. 😅

**Review by ademarCardoso — APPROVED** [review]
It's a great contribution to the code base, thanks for the help my friend.
I will pass it on to our QA to do the final tests and in a few days I believe we can launch a release with this new feature 🚀

File: src/cli.js [hunk 4]
```diff
@@ -273,6 +273,7 @@ program
   .option('--keys <KEYS>', 'Field names in your story object which should be used for filtering. Multiple keys should separated by comma.')
   .option('--operations <OPERATIONS>', 'Operations to be used for filtering. Can be: is, in, not_in, like, not_like, any_in_array, all_in_array, gt_date, lt_date, gt_int, lt_int, gt_float, lt_float. Multiple operations should be separated by comma.')
   .option('--values <VALUES>', 'Values to be used for filtering. Any string or number. If you want to use multiple values, separate them with a comma. Multiple values should be separated by comma.')
+  .option('--c-groups <UUIDs>', 'Synchronize components based on their group UUIDs separated by commas')
```
Comment (thiagosaife) [inline src/cli.js]: @CaprinaeWT Please change `--c-groups` to `--components-groups`. We do prefer the names as descriptive as possible. Thank you.

File: src/cli.js [hunk 5]
```diff
@@ -289,9 +290,12 @@ program
         filter,
         keys,
         operations,
-        values
+        values,
+        cGroups
```
Comment (thiagosaife) [inline src/cli.js]: @CaprinaeWT same here. From `cGroups` to `componentsGroups`.

**Review by thiagosaife — CHANGES_REQUESTED** [review]
I left some comments about naming changes.

**Review by thiagosaife — APPROVED** [review]

**Pull request merged**